joblib/_parallel_backends.py

```diff
@@ -411,6 +422,8 @@ def terminate(self):
             delete_folder(self._workers._temp_folder)
             self._workers = None
 
+        self.reset()
```

Comment:
You will need to call reset in the multiprocessing backend as well. Please add a test for backend='multiprocessing' too.

Reply:
I did that initially but other tests were then failing. I pushed another that tries to address the problems raised but now it fails on Windows (timeout)